pubpub/pubpub
# Add "Pub Connections" views to Pub body

This PR (finally) adds Pub connections to the Pub body, including a new query for sibling Pub edges. There is also a new `canViewPubEdge` function that gets called to avoid leaking information about unreleased Pubs. It doesn't quite fully determine whether a user can see a given Pub if it's outside of the current Community scope, but it's good enough to handle intra-community linking without being too heavyweight.

This needs extensive testing and I'm pretty certain I missed something somewhere, so feel free to try to break it!

_Some screenshots:_

Pub connections at the top of a Pub:
<img width="1034" alt="Screen Shot 2020-07-10 at 4 50 38 PM" src="https://user-images.githubusercontent.com/2208769/87202609-e50b5c80-c2ce-11ea-971f-5ad8123eb992.png">

Pub connections at the bottom of a Pub (you'll have to take my word for it):
<img width="700" alt="Screen Shot 2020-07-10 at 4 50 51 PM" src="https://user-images.githubusercontent.com/2208769/87202616-e8064d00-c2ce-11ea-962d-e18a72fb5956.png">

Same as above, but showing sibling objects.
<img width="715" alt="Screen Shot 2020-07-10 at 4 51 17 PM" src="https://user-images.githubusercontent.com/2208769/87202617-e8064d00-c2ce-11ea-98e0-6007605329ca.png">

_Test plan:_

- Make `B` a reply to `A` and verify that `A` shows up at the top of `B` and `B` shows up at the bottom of `A`
- Make `C` a reply to `A` and verify that `B` and `C` reference each other as siblings
- Create an unreleased Pub `D` linked to `A` and verify that it appears on `A` iff you are logged in.
